src/main/java/com/github/junrar/unpack/vm/VMStandardFilters.java

```diff
@@ -73,6 +73,9 @@ public static VMStandardFilters findFilter(int filter) {
         if (VMSF_DELTA.equals(filter)) {
             return VMSF_DELTA;
         }
+        if (VMSF_UPCASE.equals(filter)) {
```

Comment:
Should we instead do something like:

```java
for (VMStandardFilters value : VMStandardFilters.values()) {
  if (value.equals(filter)) return value;
}
```

which would be future proof if ever we add a new value in the enum?

Reply:
This could be done for the enums as `VMCommands` is already implemented like this. But I like to mention that this might have a performance impact if the parse function is executed in a hot loop. Since `values()` will make a copy of all the enums for every invocation there might be increased GC pressure with a change like that. This could be avoided by keeping a static array of the values in the class which also would increase the memory needed.

Additionally I don't think it is really necessary to change the function like that after adding the tests. The `ParameterizedTest` with the `EnumSource` ensures that we test the parse function for every declared value of the enum. If someone adds a new value and does not change the parse function the test would then fail. Also I don't think thoes values will change frequently since they are defined by the rar format.

Reply:
fair enough